app/backend/Services/ReadRetrieveReadChatService.cs

```diff
@@ -244,4 +254,323 @@ You answer needs to be a json object with the following format.
 
         return new ChatAppResponse(new[] { choice });
     }
-}
+
+    public async Task ReplyStreamingAsync(
```

Comment:
Seems many codes here are duplicated with the method above? We shouldn't copy but extract and reuse

Reply:
Hi Chenyang,

The reason I tried to split the two functions is to avoid affecting the current non-streaming flow.
There are differences between the two functions:

- Non-streaming has a return value, while streaming does not.
- Parameters are different.
- In streaming, I use a different format to ask the LLM instead of JSON. This is because, with JSON, streaming might contain incomplete JSON structures (e.g., `{"answer": "answer which might not be completed ...`), which would look strange when displayed to customers.

I think we can wait to get feedback from maintainer's repo for this function as well.